app/models/pawn.rb

```diff
@@ -1,3 +1,33 @@
 class Pawn < Piece
 
+  def is_valid?(new_x, new_y)
+    return false unless super   
+    return false unless allowed_move?(new_x, new_y)
+    return false if opponent_here?(new_x, new_y) and not allowed_capture?(new_x, new_y)
+    return true
+  end
+
+  def allowed_capture?(new_x, new_y)
+    white ? allowed_diagonal = 1 : allowed_diagonal = -1
+    return opponent_here?(new_x, new_y) && (new_y - y_coordinate) == allowed_diagonal && (new_x - x_coordinate).abs == 1 
+  end
+
+  def allowed_move?(new_x, new_y)
+    white ? allowed_moves = [1] : allowed_moves = [-1]
+    allowed_moves.push(allowed_moves[0]*2) if first_move?()
+    return allowed_moves.include?(new_y - y_coordinate) && (new_x - x_coordinate).abs <= 1
+  end
+
+  def first_move?()
+    return (white && y_coordinate == 2) || ((not white) && y_coordinate == 7) 
+  end
+
+  def opponent_here?(new_x, new_y)

```

Comment:
This method and own_piece_here? share almost the same logic, right? Maybe we can leave one of them and make the switch on empty_here? and if not, its one of two cases: own piece or opponents piece.

Reply:
Ok so I guess empty_here? would return false if there's no piece, but not sure what it should return if there is own or opponents piece there.